kong/api/crud_helpers.lua

```diff
@@ -59,6 +59,25 @@ function _M.find_upstream_by_name_or_id(self, dao_factory, helpers)
   end
 end
 
+function _M.find_target_by_target_or_id(self, dao_factory, helpers)
+  local filter_keys = {
+    upstream_id = self.upstream.id,
+    [utils.is_valid_uuid(self.params.target_or_id) and "id" or "target"] = self.params.target_or_id
+  }
+  self.params.target_or_id = nil
+
+  local rows, err = dao_factory.targets:find_all(filter_keys)
+  if err then
+    return helpers.yield_error(err)
+  end
+
+  -- We know target and id are unique, so if we have a row, it must be the only one
```

Comment:
that does not matter, because all of those records might differ on the `weight` property, but they will all have the same `target` field, and thats the field you're looking for.